File: DotNet/Opertat-Core/Brain Layers/Interfaces/IConduction.cs
using System;
using MathNet.Numerics.LinearAlgebra;

namespace Photon.NeuralNetwork.Opertat.Implement
{
    public interface IConduction
    {
        int ExtraCount { get; }
        double CertaintyFactor { get; set; }
        Matrix<double> Regularize(Matrix<double> synapse);
        Vector<double> Conduct(Vector<double> signal);
        Vector<double> Conduct(NeuralNetworkFlash flash, int layer);
        Vector<double> ConductDerivative(NeuralNetworkFlash flash, int layer);
    }
}

File: Java/Opertat-Core/src/Photon/NeuralNetwork/Opertat/Implement/IConduction.java
package Photon.NeuralNetwork.Opertat.Implement;

import org.ejml.data.DMatrixRMaj;

import Photon.NeuralNetwork.Opertat.NeuralNetworkFlash;

public interface IConduction {
	
	public int ExtraCount();
	public DMatrixRMaj Conduct(DMatrixRMaj signal);
	public DMatrixRMaj Conduct(NeuralNetworkFlash flash, int layer);
	public DMatrixRMaj ConductDerivative(NeuralNetworkFlash flash, int layer);
	
}
